scene/gui/text_edit.cpp

```diff
@@ -4504,6 +4560,68 @@ int TextEdit::get_caret_count() const {
 	return carets.size();
 }
 
+void TextEdit::insert_caret_at_carets(bool p_below) {
+	for (const int &caret_index : caret_index_edit_order) {
+		const int caret_line = get_caret_line(caret_index);
+		const int caret_column = get_caret_column(caret_index);
+
+		// The last fit x will be cleared if the caret has a selection,
+		// but if it does not have a selection the last fit x will be
+		// transferred to the new caret
+		int caret_from_column = 0, caret_to_column = 0, caret_last_fit_x;
+		if (has_selection(caret_index)) {
+			// If the selection goes over multiple lines, deselect it.
+			if (get_selection_from_line(caret_index) != get_selection_to_line(caret_index)) {
+				deselect(caret_index);
+				caret_last_fit_x = carets[caret_index].last_fit_x;
+			} else {
+				caret_from_column = get_selection_from_column(caret_index);
+				caret_to_column = get_selection_to_column(caret_index);
+				caret_last_fit_x = -1;
+				carets.write[caret_index].last_fit_x = _get_column_x_offset_for_line(caret_column, caret_line, caret_column);
+			}
+		} else {
+			caret_last_fit_x = carets[caret_index].last_fit_x;
+		}
```

Comment:
`Nit` Can set `caret_last_fit_x` to get rid of the else:
```suggestion
		int caret_from_column = 0, caret_to_column = 0;
		int caret_last_fit_x = carets[caret_index].last_fit_x;
		if (has_selection(caret_index)) {
			// If the selection goes over multiple lines, deselect it.
			if (get_selection_from_line(caret_index) != get_selection_to_line(caret_index)) {
				deselect(caret_index);
			} else {
				caret_from_column = get_selection_from_column(caret_index);
				caret_to_column = get_selection_to_column(caret_index);
				caret_last_fit_x = -1;
				carets.write[caret_index].last_fit_x = _get_column_x_offset_for_line(caret_column, caret_line, caret_column);
			}
		}
```